client/my-sites/stats/features/modules/stats-top-posts/stats-top-posts.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+	isRequestingSiteStatsForQuery,
+	getSiteStatsNormalizedData,
+} from 'calypso/state/stats/lists/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import EmptyModuleCard from '../../../components/empty-module-card/empty-module-card';
+import EmptyStateAction from '../../../components/empty-state-action';
+import { SUPPORT_URL } from '../../../const';
+import StatsModule from '../../../stats-module';
+import StatsModulePlaceholder from '../../../stats-module/placeholder';
+
+type StatsTopPostsProps = {
+	className?: string;
+	period: string;
+	query: string;
+	moduleStrings: {
+		title: string;
+		item: string;
+		value: string;
+		empty: string;
+	};
+};
+
+// TODO: move to a shared file if this is the final URL
+const JETPACK_SUPPORT_AI_URL =
+	'https://jetpack.com/support/jetpack-blocks/jetpack-ai-assistant-block/';
+const JETPACK_SUPPORT_SOCIAL_URL = 'https://jetpack.com/support/jetpack-social/';
+
+const StatsTopPosts: React.FC< StatsTopPostsProps > = ( {
+	period,
+	query,
+	moduleStrings,
+	className,
+} ) => {
+	const translate = useTranslate();
+	const siteId = useSelector( getSelectedSiteId ) as number;
+	const statType = 'statsTopPosts';
+
+	// TODO: sort out the state shape.
+	const requesting = useSelector( ( state: any ) =>
+		isRequestingSiteStatsForQuery( state, siteId, statType, query )
+	);
+	const data = useSelector( ( state ) =>
+		getSiteStatsNormalizedData( state, siteId, statType, query )
+	) as [ id: number, label: string ]; // TODO: get post shape and share in an external type file.
+
+	const cardActions = [
+		{
+			icon: starEmpty,
+			text: translate( 'Craft engaging content with Jetpack AI assistant' ),
+			onClick: () => {
+				// analytics event tracting handled in EmptyStateAction component
+
+				setTimeout( () => ( window.location.href = localizeUrl( JETPACK_SUPPORT_AI_URL ) ), 250 );
+			},
+			analyticsDetails: {
+				from: 'module_top_posts',
+				feature: 'ai_assistant',
+			},
+		},
+		{
+			icon: megaphone,
+			text: translate( 'Share on social media with one click' ),
+			onClick: () => {
+				// analytics event tracting handled in EmptyStateAction component
+
+				setTimeout(
+					() => ( window.location.href = localizeUrl( JETPACK_SUPPORT_SOCIAL_URL ) ),
+					250
+				);
+			},
+			analyticsDetails: {
+				from: 'module_top_posts',
+				feature: 'social_sharing',
+			},
+		},
+	];
+
+	return (
+		<>
+			{ /* This will be replaced with ghost loaders, fallback to the current implementation until then. */ }
+			{ requesting && <StatsModulePlaceholder isLoading={ requesting } /> }
+			{ ( ! data || ! data?.length ) && (
+				<StatsCard
+					className={ className }
+					title={ moduleStrings.title }
+					isEmpty
+					emptyMessage={
+						<EmptyModuleCard
+							icon={ trendingUp }
+							description={ translate(
+								'Your top {{link}}posts and pages{{/link}} will display here and learn what content resonates the most. Start creating and sharing!',
```

Comment:
This sentence is a little awkward. It seems like the second part is missing something. Should this read:

> Your top {{link}}posts and pages{{/link}} will display here and **you can** learn what content resonates the most

Reply:
That's a good point. I like the suggestion.